buffer.go

```diff
@@ -0,0 +1,159 @@
+package stats
+
+import (
+	"io"
+	"sync"
+	"sync/atomic"
+	"time"
+	"unsafe"
+)
+
+// Buffer is the implementation of a measure handler which uses a Serializer to
+// serialize the metric into a memory buffer and write them once the buffer has
+// reached a target size.
+type Buffer struct {
+	// Target size of the memory buffer where metrics are serialized.
+	//
+	// If left to zero, a size of 1024 bytes is used as default (this is low,
+	// you should set this value).
+	BufferSize int
+
+	// The Serializer used to write the measures.
+	//
+	// This field cannot be nil.
+	Serializer Serializer
+
+	buffer unsafe.Pointer
```

Comment:
Gotcha.  So one could say that this buffer is all-or-nothing?

Asking because I've used either a circular buffer implementation or just didn't serialize and used a typed slice buffer.